File: backend-api/Controllers/CharacterController.cs
using backend_api.Models;
using backend_api.Services;
using Microsoft.AspNetCore.Mvc;

namespace backend_api.Controllers;

[ApiController]
[Route("[controller]")]
public class CharacterController : ControllerBase
{
  private ILogger<CharacterController> Logger { get; }
  private ICharacterService CharacterService { get; }

  public CharacterController(ILogger<CharacterController> logger,
                             ICharacterService characterService)
  {
    Logger = logger;
    CharacterService = characterService;
  }

  [HttpGet(Name = "GetCharacter")]
  public IEnumerable<Character> GetAll()
  {
    Logger.LogInformation($"GET characters");
    return CharacterService.GetCharacters();
  }

  [HttpGet("{id}", Name = "GetCharacterById")]
  public Character Get(string id)
  {
    Logger.LogInformation($"GET character:${id}");
    return CharacterService.GetCharacter(Guid.Parse(id));
  }
}

File: frontend-api/src/controllers/CharacterController.ts
import { Controller, Get, Param } from "routing-controllers";

@Controller()
export class CharacterController {
  @Get("/character")
  public async getAll() {
    const response = await fetch(`${process.env.BACKEND_API}/character`);
    return await response.json();
  }

  @Get("/character/:id")
  public async getOne(@Param("id") id: string) {
    const response = await fetch(`${process.env.BACKEND_API}/character/${id}`);
    return await response.json();
  }
}
